doc/source/howtos/encrypt-traffic.rst

```diff
@@ -236,4 +236,46 @@ assumig there are two nodes in the cluster:
 
       wsrep_provider_options=socket.ssl=yes;socket.ssl_ca=/path/to/upgrade-ca.pem;socket.ssl_cert=/path/to/new-cert.pem;socket.ssl_key=/path/to/new-key.pem
 
+.. _ssl-auto-conf:
+
+SSL Automatic Configuration
+===========================
+
+|PXC| includes the :variable:`pxc-encrypt-cluster-traffic` variable
+that enables automatic configuration of SSL encrytion.
+By default, it is disabled, meaning that you need to configure SSL manually
+if you want to encrypt SST and IST traffic, and internal communication.
+
+This variable cannot be changed on the command line during runtime.
+To enable automatic configuration of SSL encryption,
+set ``pxc-encrypt-cluster-traffic=ON`` in the :file:`my.cnf` file.
+This will override encryption settings, but not other options related to SST,
+such as compression.
+
+When you enable automatic configuration of SSL encryption,
```

Comment:
Preliminaries
The usage of SSL within PXC requires that the SSL certificates/key files all be the same (or generated from the same CA).  So, before turning on SSL auto-configuration, it is necessary to create the appropriate SSL certs/keys and copy them to the nodes before startup. The location of the CA, certificate, and key files are specified with the 'ssl-ca', 'ssl-cert', and 'ssl-key' options respectively (in the [mysqld] section). If these options are not specified, then the files are assumed to be in the data directory as 'ca.pem', 'server-cert.pem', and 'server-key.pem' respectively.
(link to the section where we talked about how to generate the SSL file)

If the files are kept in their default location in the datadir, then the only option that needs to be set is 'pxc-encrypt-cluster-traffic=ON'.

PXC SSL auto-configuration occurs in two steps:
(1) The SSL files are located.
PXC first looks for the 'ssl-ca', 'ssl-cert' and 'ssl-key' options in the my.cnf file.  If these are not found, then the default names in the datadir ('ca.pem', 'server-cert.pem', and 'server-key.pem') files are used.  Note that the '[sst]' section is not searched.

(2) The Galera and SST configuration settings are overridden.
To ensure proper functioning, the Galera/SST options are then overridden to use these values for the SSL files. Other SSL settings are unchanged. The effect of the SSL auto-configuration is equivalent to setting the my.cnf as such:

[mysqld]
wsrep_provider_options=”socket.ssl_key=server-key.pem;socket.ssl_cert=server-cert.pem;socket.ssl_ca=ca.pem”

[sst]
encrypt=4
ssl-key=server-key.pem
ssl-ca=ca.pem
ssl-cert=server-cert.pem